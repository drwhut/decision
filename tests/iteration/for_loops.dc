Start~#1

< Counting from positive integer to positive integer in steps of positive integer >
For(#1, 1, 10, 1)~#2, #3, #4
Print(#2, #3)
Print(#4, "done")~#5

< Counting from positive integer to positive integer in steps of negative integer >
For(#5, 10, 1, -1)~#6, #7, #8
Print(#6, #7)
Print(#8, "done")~#9

< Counting from positive integer to negative integer in steps of positive integer >
For(#9, 5, -5, 1)~#10, #11, #12
Print(#10, #11)
Print(#12, "done")~#13

< Counting from positive integer to negative integer in steps of negative integer >
For(#13, 5, -5, -1)~#14, #15, #16
Print(#14, #15)
Print(#16, "done")~#17

< Counting from negative integer to positive integer in steps of positive integer >
For(#17, -5, 5, 2)~#18, #19, #20
Print(#18, #19)
Print(#20, "done")~#21

< Counting from negative integer to positive integer in steps of negative integer >
For(#21, -5, 5, -1)~#22, #23, #24
Print(#22, #23)
Print(#24, "done")~#25

< Counting from negative integer to negative integer in steps of positive integer >
For(#25, -10, -5, 1)~#26, #27, #28
Print(#26, #27)
Print(#28, "done")~#29

< Counting from negative integer to negative integer in steps of negative integer >
For(#29, -1, -10, -1)~#30, #31, #32
Print(#30, #31)
Print(#32, "done")~#33

< Counting from positive float to positive float in steps of positive float >
For(#33, 0.0, 10.0, 0.5)~#34, #35, #36
Print(#34, #35)
Print(#36, "done")~#37

< Counting from positive float to positive float in steps of negative float >
For(#37, 5.0, 1.0, -0.5)~#38, #39, #40
Print(#38, #39)
Print(#40, "done")~#41

< Counting from positive float to negative float in steps of positive float >
For(#41, 1.0, -1.0, -0.25)~#42, #43, #44
Print(#42, #43)
Print(#44, "done")~#45

< Counting from positive float to negative float in steps of negative float >
For(#45, 10.0, -10.0, -2.5)~#46, #47, #48
Print(#46, #47)
Print(#48, "done")~#49

< Counting from negative float to positive float in steps of positive float >
For(#49, -1000.0, 1000.0, 125.25)~#50, #51, #52
Print(#50, #51)
Print(#52, "done")~#53

< Counting from negative float to positive float in steps of negative float >
For(#53, -1.0, 1.0, -0.5)~#54, #55, #56
Print(#54, #55)
Print(#56, "done")~#57

< Counting from negative float to negative float in steps of positive float >
For(#57, -100.0, -10.0, 12.5)~#58, #59, #60
Print(#58, #59)
Print(#60, "done")~#61

< Counting from negative float to negative float in steps of negative float >
For(#61, -10.0, -100.0, -20.125)~#62, #63, #64
Print(#62, #63)
Print(#64, "done")~#65

< Counting with a mixture of positive integers and floats >
For(#65, 1, 10, 0.5)~#66, #67, #68
Print(#66, #67)
Print(#68, "done")~#69