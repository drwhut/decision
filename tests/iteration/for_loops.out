1
2
3
4
5
6
7
8
9
10
done
10
9
8
7
6
5
4
3
2
1
done
done
5
4
3
2
1
0
-1
-2
-3
-4
-5
done
-5
-3
-1
1
3
5
done
done
-10
-9
-8
-7
-6
-5
done
-1
-2
-3
-4
-5
-6
-7
-8
-9
-10
done
0
0.5
1
1.5
2
2.5
3
3.5
4
4.5
5
5.5
6
6.5
7
7.5
8
8.5
9
9.5
10
done
5
4.5
4
3.5
3
2.5
2
1.5
1
done
1
0.75
0.5
0.25
0
-0.25
-0.5
-0.75
-1
done
10
7.5
5
2.5
0
-2.5
-5
-7.5
-10
done
-1000
-874.75
-749.5
-624.25
-499
-373.75
-248.5
-123.25
2
127.25
252.5
377.75
503
628.25
753.5
878.75
done
done
-100
-87.5
-75
-62.5
-50
-37.5
-25
-12.5
done
-10
-30.125
-50.25
-70.375
-90.5
done
1
1.5
2
2.5
3
3.5
4
4.5
5
5.5
6
6.5
7
7.5
8
8.5
9
9.5
10
done
